vcv-rack arm

luca3s opened the pull request:
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---
Added the Arm Version to VCV-Rack. It's my first time contributing to homebrew so I hope you can excuse the mess I made with the commits.

bevanjkay commented:
/rebase

bevanjkay commented:
@luca3s Thank you for submitting this. Please try to add a [zap stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

luca3s commented:
@bevanjkay hi, I added the zap stanza. I added x64 and arm64 user folders to it as users have both when switching. I also added the old and the new user folder locations. There I am not sure if it is correct/needed. But the current 2.5.0 release is the one where the switch happened, so I thought that some users probably still have the old location. see: https://github.com/VCVRack/Rack/blob/v2/CHANGELOG.md

krehel commented:
/rebase

krehel commented on Casks/v/vcv-rack.rb:
```diff
@@ -12,7 +15,20 @@
   uninstall pkgutil: "com.vcvrack.rack"
+
+  zap trash: [
+    "~/Documents/Rack2/*.json",
+    "~/Documents/Rack2/autosave",
+    "~/Documents/Rack2/log.txt",
+    "~/Documents/Rack2/plugins-mac-arm64",
+    "~/Documents/Rack2/plugins-mac-x64",
+    "~/Library/Application Support/Rack2/*.json",
+    "~/Library/Application Support/Rack2/autosave",
+    "~/Library/Application Support/Rack2/log.txt",
+    "~/Library/Application Support/Rack2/plugins-mac-arm64",
+    "~/Library/Application Support/Rack2/plugins-mac-x64",
```
This can be simplified down to directories.  All files inside will also be removed.

```suggestion
    "~/Documents/Rack2",
    "~/Library/Application Support/Rack2",
```

luca3s replied:
This doesn't work as Patches the user makes are by default also stored in a subfolder called patches. And as far as I understood the zap requirements user created files shouldn't be removed.

krehel replied:
Correct.  Which was a follow-up question on including the `~/Documents/Rack2` folder in this to start with.  This is fine then, just a bit less than ideal.  But appreciate the work done here!

luca3s replied:
Thank you :)
